### PR TITLE
Improving online RL logging

### PR DESCRIPTION
**Pull request by abaheti95**
- Adds `prompt_id` and `reward` logging to mlflow table.
- Adds entropy logging during training.

Working run `mcli logs test-logging-pr-a96D3V -f`

### CODE REVIEW

File: compose_rl/ppo/callback.py [hunk 1]
```diff
@@ -642,6 +654,8 @@ def _interact_with_env(self, batch: dict[str, torch.Tensor]):
                 # Determine the number of generating calls we want to make
                 # We can have the generate size be greater than the device train microbatch size
                 num_gen_calls = self.num_batches_per_update * self.device_train_batch_size // self.device_generate_batch_size
+                if num_gen_calls <= 0:
```
Comment (gupta-abhay) [inline compose_rl/ppo/callback.py:657]: why will we ever hit this? It just means that the settings were incorrect in the yaml right?

File: compose_rl/utils/utils.py [hunk 2]
```diff
@@ -941,6 +959,24 @@ def get_log_probs_from_logits(logits: torch.Tensor, actions: torch.Tensor):
     return logpy
 
 
+def get_entropies_from_logits(logits: torch.Tensor, actions: torch.Tensor):
```
Comment (gupta-abhay) [inline compose_rl/utils/utils.py:962]: why is entropy useful to log - asking for me ?

File: compose_rl/utils/utils.py [hunk 3]
```diff
@@ -941,6 +959,24 @@ def get_log_probs_from_logits(logits: torch.Tensor, actions: torch.Tensor):
...
+    actions_probs = torch.gather(pd, 2, actions.unsqueeze(2).long()).squeeze(-1)
+
+    # Calculate entropy for those specific actions: -p*log(p)
+    # Adding small epsilon to avoid log(0)
+    pointwise_entropies = -actions_probs * torch.log(actions_probs + 1e-10)
+
+    # Mean over sequence length (dim=1) to get one entropy value per sequence
+    sequence_entropies = torch.mean(pointwise_entropies, dim=1)
+
+    return sequence_entropies
```
Comment (gupta-abhay) [inline compose_rl/utils/utils.py:977]: return the `torch.mean` directly.

File: compose_rl/utils/utils.py [hunk 4]
```diff
@@ -106,6 +106,24 @@ def get_log_probs(
...
 
+def get_entropies(
+    logits: torch.Tensor,
+    actions: torch.Tensor,
+    prompt_len: torch.Tensor,
+    max_gen_len: Union[torch.Tensor, int],
+) -> torch.Tensor:
+    """Gets the entropies from the generated logits.
+
+    Inputs:
```
Comment (gupta-abhay) [inline compose_rl/utils/utils.py:117]: Follow docformatter convention of `Args` and `Returns`

File: compose_rl/utils/utils.py (hunk 2 above)
Comment (gupta-abhay) [inline compose_rl/utils/utils.py:962]: add correct docstrings for the fn and then also assign return type `-> torch.Tensor`

**Review by gupta-abhay — APPROVED** [review]
lgtm - small nits !

File: compose_rl/ppo/callback.py (hunk 1 above)
Comment (abaheti95) [reply compose_rl/ppo/callback.py:657]: Yeah... This is mainly to flag when the YAML is wrong. Can remove it if it looks unnecessary.

File: compose_rl/utils/utils.py (hunk 2 above)
Comment (abaheti95) [reply compose_rl/utils/utils.py:962]: Want to track if the generations over time show entropy collapse or blowup as we are training.
